src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -1247,6 +1247,12 @@ public SolutionCompilationState WithOptions(SolutionOptionSet options)
             this.SolutionState.WithOptions(options));
     }
 
+    /// <summary>
+    /// Updates entries in our <see cref="_sourceGeneratorExecutionVersionMap"/> to the corresponding values in the
+    /// given <paramref name="sourceGeneratorExecutionVersions"/>.  Importantly, <paramref
+    /// name="sourceGeneratorExecutionVersions"/> must refer to projects in this solution.  Projects not mentioned in
+    /// <paramref name="sourceGeneratorExecutionVersions"/> will not be touched (and they will stay in the map).
+    /// </summary>
     public SolutionCompilationState WithSourceGeneratorExecutionVersions(
```

Comment:
@ToddGrun would you prefer the name of this method be changed.  'With' tends to imply a total overwrite.  But those aren't the semantics here.  Maybe `UpdateSpecificSourceGenerationExecutionVersions`

Reply:
Yeah, Update probably fits a bit better here

Reply:
done.